src/clj/ataru/applications/application_store.clj

```diff
@@ -481,15 +481,40 @@
    :email         email
    :hakukohteet   hakukohde})
 
-(defn get-applications-by-haku
+
+(defn- payment-obligation-to-application [application payment-obligations]
+  (if-let [obligation (->> (get payment-obligations (:oid application))
+                           (group-by :hakukohde)
+                           (reduce (fn [r [k v]]
+                                     (assoc r k (-> v first :state)))
+                                   {})
+                           (not-empty))]
+    (assoc application :payment-obligations obligation)
+    application))
+
+(defn- get-external-applications
   [haku-oid hakukohde-oid hakemus-oids]
-  (->> (exec-db :db yesql-applications-by-haku-and-hakukohde-oids {:haku_oid       haku-oid
-                                                                   ; Empty string to avoid empty parameter lists
-                                                                   :hakukohde_oids (cond-> [""]
-                                                                                           (some? hakukohde-oid)
-                                                                                           (conj hakukohde-oid))
-                                                                   :hakemus_oids   (cons "" hakemus-oids)})
-       (map unwrap-vts-application)))
+  (->> (exec-db :db
+                yesql-applications-by-haku-and-hakukohde-oids
+                {:haku_oid       haku-oid
+                 ; Empty string to avoid empty parameter lists
+                 :hakukohde_oids (cond-> [""]
+                                         (some? hakukohde-oid)
+                                         (conj hakukohde-oid))
+                 :hakemus_oids   (cons "" hakemus-oids)})
+       (map unwrap-external-application)))
+
+(defn- payment-obligations-for-applications [hakemus-oids]
+  (->> (exec-db :db
+                yesql-get-payment-obligation-for-applications
+                {:hakemus_oids (cons "" hakemus-oids)})
```

Comment:
Tässä ei kai oo tarvetta tolla `(cons "" hakemus-oids)` tempulle, kun aina pitäis olla `hakemus-oid`:ja.